shared/src/processFileEdits.ts

```diff
@@ -0,0 +1,23 @@
+export type FileEdit = {
+  start: number;
+  end: number;
+  newText: string;
+};
+
+export const processFileEdits = (
+  oldText: string,
+  fileEdits: FileEdit[],
+): string => {
+  const fileEditsSortedByStart = fileEdits.sort((a, b) => b.start - a.start);
```

Comment:
I think this won't be a performance concern here and IMHO using an immutable update makes it usually easier to debug things as we can always inspect the previous state of a structure in the call stack etc
```suggestion
  const fileEditsSortedByStart = [...fileEdits].sort((a, b) => b.start - a.start);
```